src/main/java/org/spongepowered/api/event/item/inventory/ClickInventoryEvent.java

```diff
@@ -61,6 +63,29 @@
         interface Middle extends Drag, ClickInventoryEvent.Middle {}
     }
 
+    /**
+     * Fires when the client requests a recipe to be crafted.
+     */
+    interface Recipe extends ClickInventoryEvent {
+
+        /**
+         * Returns the requested recipe-
+         *
+         * @return The requested recipe.
+         */
+        CraftingRecipe getRecipe();
```

Comment:
This should be changed to `Recipe` instead, in 1.13, `FurnaceRecipe`s are also used in a recipe book.